stl/inc/xstring

```diff
@@ -3053,52 +3047,51 @@ private:
+                auto&& _Alproxy       = _GET_PROXY_ALLOCATOR(_Alty, _Al);
+                auto&& _Right_alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Right_al);
+                _Container_proxy_ptr<_Alty> _New_proxy(_Right_alproxy, _Leave_proxy_unbound{}); // throws
 
-        if (_Right._Mypair._Myval2._Large_string_engaged()) {
-            const auto _New_size     = _Right._Mypair._Myval2._Mysize;
-            const auto _New_capacity = _Calculate_growth(_New_size, 0, _Right.max_size());
-            auto _Right_al_non_const = _Right_al;
-            const auto _New_ptr      = _Right_al_non_const.allocate(_New_capacity); // throws
+                if (_Right._Mypair._Myval2._Large_string_engaged()) {
+                    const auto _New_size     = _Right._Mypair._Myval2._Mysize;
+                    const auto _New_capacity = _Calculate_growth(_New_size, 0, _Right.max_size());
+                    auto _Right_al_non_const = _Right_al;
+                    const auto _New_ptr      = _Right_al_non_const.allocate(_New_capacity); // throws
 
 #if _HAS_CXX20
-            if (_STD is_constant_evaluated()) { // Begin the lifetimes of the objects before copying to avoid UB
-                _Traits::assign(_Unfancy(_New_ptr), _New_size + 1, _Elem());
-            }
+                    if (_STD is_constant_evaluated()) { // Begin the lifetimes of the objects before copying to
+                                                        // avoid UB
+                        _Traits::assign(_Unfancy(_New_ptr), _New_size + 1, _Elem());
+                    }
 #endif // _HAS_CXX20
 
-            _Traits::copy(_Unfancy(_New_ptr), _Unfancy(_Right._Mypair._Myval2._Bx._Ptr), _New_size + 1);
-            _Tidy_deallocate();
-            _Mypair._Myval2._Bx._Ptr = _New_ptr;
-            _Mypair._Myval2._Mysize  = _New_size;
-            _Mypair._Myval2._Myres   = _New_capacity;
-        } else {
-            _Copy_assign_val_from_small(_Right);
-        }
-
-        _Pocca(_Al, _Right_al);
-        _New_proxy._Bind(_Alproxy, _STD addressof(_Mypair._Myval2));
-    }
+                    _Traits::copy(_Unfancy(_New_ptr), _Unfancy(_Right._Mypair._Myval2._Bx._Ptr), _New_size + 1);
+                    _Tidy_deallocate();
+                    _Mypair._Myval2._Bx._Ptr = _New_ptr;
+                    _Mypair._Myval2._Mysize  = _New_size;
+                    _Mypair._Myval2._Myres   = _New_capacity;
+                } else {
+                    _Copy_assign_val_from_small(_Right);
+                }
 
-public:
-    _CONSTEXPR20 basic_string& operator=(const basic_string& _Right) {
-        if (this != _STD addressof(_Right)) {
-            _Copy_assign(_Right, _Choose_pocca<_Alty>{});
+                _Pocca(_Al, _Right_al);
+                _New_proxy._Bind(_Alproxy, _STD addressof(_Mypair._Myval2));
+            }
+        } else {
+            _Pocca(_Getal(), _Right._Getal());
+            assign(_Right._Mypair._Myval2._Myptr(), _Right._Mypair._Myval2._Mysize);

```

Comment:
I think we'd prefer to be consistent with the fallthrough design the other headers used, by:
* returning early on 3091
* moving 3060-3061 outside of the branching, down to 3096,
* dropping this branch (3092-3095)
That would save a couple of lines of duplication, but the consistency with the other headers is really the important thing.

Reply:
Done,
Thoughts about `<xhash>`, which is inconsistent?
 * Can keep as is
 * Can move checks to a destructor of a helper class. This will change the behavior a bit: checks would be performed on exception either. The change might be good, but currently the PR tries to have no behavior changes.
 * Introduce inner function for early return
 * `goto exit`
 * duplicate checks in early return cases (the worst alternative, I think)

Reply:
For my money, keep-as-is in this PR. Our goal is simply to transform tag dispatch into `if constexpr`, and all of these options feel more like refactoring than rearranging.